crates/egui/src/context.rs

```diff
@@ -113,13 +113,13 @@ impl ContextImpl {
         let viewport = self.viewports.entry(viewport_id).or_default();
 
         viewport.repaint.prev_frame_paint_delay = viewport.repaint.repaint_delay;
+        viewport.repaint.outstanding = viewport.repaint.outstanding.saturating_sub(1);
```

Comment:
The point of `outstanding` it to run two refreshes on `request_repaint`, and on startup, but with this PR that is now ignored.

That said, thanks to https://github.com/emilk/egui/pull/3930 we might not need that "repaint twice" logic

Reply:
Ah okay, I didn't realize that you actually wanted two invocations. It does make sense though, as it serves as a simple form of resilience against all kinds of "frame updates stuff and then expects it to show up in the next repaint" -- let's try to keep that.

Note however that the previous logic didn't actually work: the `== 0` branch was previously never taken at all, because `begin_frame_repaint_logic` isn't ever actually called once `outstanding` reached 0. You can verify this by throwing in a breakpoint or `panic!()`.

Would it make sense to move the decrement as suggested in this PR and then simply explicitly set `outstanding` to 2 in both the constructor and `request_repaint`? I believe that this would also convey the intention of redrawing twice more clearly.

Reply:
> Ah okay, I didn't realize that you actually wanted two invocations. It does make sense though, as it serves as a simple form of resilience against all kinds of "frame updates stuff and then expects it to show up on screen soon"

Exactly. It's a guard-rail, that's is nice to have. I'm not sure it is actually needed for anything (I'm turning it off now and testing, and nothing is obviously broken). Especially with #3930 merged the need should be smaller.

---

If you go to the backend panel of `egui_demo_app` there is a simple test for some of this:

![image](https://github.com/emilk/egui/assets/1148717/df4ac85c-36e5-41ca-8432-c3c84e4c3a23)

Clicking that button once will produce exactly 2 frames to be repainted after a five second delay, so it's not _all_ broken.

Reply:
Hmm yeah, true. If you just call `request_repaint_after` once, it is working as intended. However, if you call it on every frame (like my sample code / diff does) in order to request a redraw e.g. at least once per second, it falls over.